client/src/components/Search/Search.js

```diff
@@ -32,30 +68,84 @@ const Search = () => {
           >
             Гости
             <div
-              className={`flex flex-col absolute search-guests mt-5 p-5 w-72 left-0 right-0 z-30 bg-white text-grey ${
+              className={`flex flex-col absolute search-guests pointer-events-none mt-5 p-5 w-72 left-0 right-0 z-30 bg-white text-grey ${
                 hidden ? "hidden" : "flex"
               }`}
             >
-              <div className="flex w-full border-black">
+              <div className="flex w-full">
                 <div className="w-2/3">Взрослые</div>
-                <div className="flex w-1/3">
-                  <div className="flex justify-center items-center text-2xl search-btn-circle cursor-pointer bg-btnGold text-white">
+                <div className="flex w-1/3 pointer-events-auto">
+                  <div
+                    className={`flex justify-center items-center text-2xl search-btn-circle outline-none bg-btnGold text-white ${
+                      adult === 0
+                        ? "opacity-50 cursor-default"
+                        : "opacity - 100"
+                    }`}
+                    onClick={(event) => {
+                      decrease("adult", event);
```

Comment:
I'd prefer 4 separate event handlers instead of `if` conditions, so you'd have `decreaseAdult`, `decreaseChildren`, etc